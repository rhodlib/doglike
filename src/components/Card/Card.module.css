.Card {
  box-sizing: border-box;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  width: 10rem;
  height: 10rem;
  background-color: var(--color-white);
  margin: 0.5rem;
  text-decoration: none;
  transition: transform 200ms ease-in-out;
  will-change: transform;
  border-radius: 4px;
  box-shadow: 0 3px 1px -2px rgba(0, 0, 0, 0.2), 0 2px 2px 0 rgba(0, 0, 0, 0.14),
    0 1px 5px 0 rgba(0, 0, 0, 0.12);
}

.Card:hover,
.Card:focus {
  transform: scale(1.1);
  box-shadow: 0 2px 4px -1px rgba(0, 0, 0, 0.2), 0 4px 5px 0 rgba(0, 0, 0, 0.14),
    0 1px 10px 0 rgba(0, 0, 0, 0.12);
  z-index: 1;
}

.CardImage {
  display: block;
  height: 3rem;
  margin-bottom: 1rem;
}

.CardNameParagraph {
  display: inline-block;
  height: 1rem;
  line-height: 1rem;
  text-transform: uppercase;
  color: var(--color-brand);
  margin: 0;
}
