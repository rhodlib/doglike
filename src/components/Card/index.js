import React from "react";
import PropTypes from "prop-types";
import styles from "./Card.module.css";
import { Link } from "react-router-dom";
import Paw from "../../images/blackpaw.png";

/**
 * Component that receives a name and make a card with a "Paw" image.
 */
export const Card = ({ dogName }) => {
  return (
    <Link to={`/post/${dogName}`} className={styles.Card}>
      <img src={Paw} className={styles.CardImage} alt="paw" />
      <p className={styles.CardNameParagraph}>{dogName}</p>
    </Link>
  );
};

Card.propTypes = {
  dogName: PropTypes.string.isRequired
};

export default Card;
