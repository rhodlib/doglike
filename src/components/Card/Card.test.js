import React from "react";
import { render } from "@testing-library/react";
import Card from "./index";
import { BrowserRouter as Router } from "react-router-dom";

it("Matches snapshot", () => {
  const { asFragment } = render(
    <Router>
      <Card dogName={"test"} />
    </Router>
  );
  expect(asFragment()).toMatchSnapshot();
});
