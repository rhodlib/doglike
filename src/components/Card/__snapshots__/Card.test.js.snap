// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot 1`] = `
<DocumentFragment>
  <a
    class="Card"
    href="/post/test"
  >
    <img
      alt="paw"
      class="CardImage"
      src="blackpaw.png"
    />
    <p
      class="CardNameParagraph"
    >
      test
    </p>
  </a>
</DocumentFragment>
`;
